File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque([(priority, key) for (key, priority) in enumerate(priorities)])   # 우선순위값과 프로세스번호(인덱스)를 튜플로 묶어 큐에 삽입.
 
    while queue:
        curPriority, curKey = queue.popleft()   # (우선순위, 프로세스번호)를 pop함.
        # cur와 배열안의 나머지숫자들과 비교.
        if all(curPriority >= item for (item, key) in queue):
            answer += 1
            # 알고싶은 위치의 프로세스 넘버라면 return.
            if curKey == location:
                return answer
        else:
            queue.append((curPriority, curKey))

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let queue = priorities.map((priority, index) => ({ priority, index}));
    let answer = 0;
    
    while (queue.length) {
        let cur = queue.shift();
        
        // 현재 프로세스의 우선순위가 더 작은게 있다면 현재프로세스는 다시 큐에 삽입
        if (queue.some(item => item.priority > cur.priority)) {
            queue.push(cur);
        } else {
            answer += 1
            if (cur.index === location) {
                return answer;
            }
        }
    }
    
}